src/runtime/hexagon_dma_context.h

```diff
@@ -0,0 +1,131 @@
+#ifndef _HALIDE_HEXAGON_DMA_CONTEXT_H_
+#define _HALIDE_HEXAGON_DMA_CONTEXT_H_
+
+#include "hexagon_dma_device_shim.h"
+
+using namespace Halide::Runtime::Internal::Qurt;
+
+/* Currently hardcoding the number of DMA engines available*/
+/* Should find a way to get this */
+/* Currently making the count to 4 as we want 1 RD Engine and 1 Write Engine per H/W Thread*/
+#define NUM_DMA_ENGINES 4
+/* Currently hardcoding the number of H/W threads available */
+/* Should find a way to get this */
+#define NUM_HW_THREADS  2
+
+#define OK 0

```

Comment:
Please consider converting this into an enum like "error_code_t" or "return_code_t" instead of freestanding macros.